src/chunkserver/op_request.cpp

```diff
@@ -871,13 +868,67 @@ void PasteChunkInternalRequest::OnApplyFromLog(std::shared_ptr<CSDataStore> data
 
 void ScanChunkRequest::OnApply(uint64_t index,
                                         ::google::protobuf::Closure *done) {
-    scanManager_->SetScanJobType(ScanType::BuildMap);
-    scanManager_->GenScanJob();
+    // read and calculate crc, build scanmap
+    uint32_t crc = 0;
+    char *readBuffer = nullptr;
+    size_t size = request_->size();
+    readBuffer = new(std::nothrow)char[size];
```

Comment:
> why not Reuse ReadChunkRequest::ReadChunk() ? or use ReadChunk as common function

Because the "ReadChunkRequest::ReadChunk()" do more things than only read data, for example it set the response but it hasn't response in 'ScanChunkRequest::ApplyFromLog'. And we need to calculate crc based readbuf, so the 'void ReadChunk()' need modified largely. If so the logic is not clear.